framework/aster-frame/src/sync/rwlock.rs

```diff
@@ -149,7 +150,7 @@ impl<T: ?Sized> RwLock<T> {
     /// in which other readers or writers waiting simultaneously will
     /// obtain the lock. Once this lock is acquired, the calling thread
     /// will not be interrupted.
-    pub fn write_irq_disabled(&self) -> RwLockWriteGuard<T> {
+    pub fn write_irq_disabled(&'a self) -> RwLockWriteGuard_<T, &'a Self> {
```

Comment:
Just a little comment about the consistency.

I've seen _at least_ three different variants of the return type for the lock guard in this PR:
 -  `pub fn callback_list(&self) -> SpinLockGuard<alloc::vec::Vec<CallbackElement>> {`
 -  `pub fn write_irq_disabled(&'a self) -> RwLockWriteGuard_<T, &'a Self> {`
 -  `pub fn try_read(&'a self) -> Option<RwMutexReadGuard<'a, T>> {`

We also have another variant that is removed in this PR:
 - `pub fn callback_list(&self) -> SpinLockGuard<'_, alloc::vec::Vec<CallbackElement>> {`

So is it better to avoid unnecessary inconsistencies here?